plugins/woocommerce-admin/client/marketplace/utils/functions.tsx

```diff
@@ -13,26 +19,28 @@ interface ProductGroup {
 }
 
 // Fetch data for the discover page from the WooCommerce.com API
-const fetchDiscoverPageData = async (): Promise< Array< ProductGroup > > => {
-	const fetchUrl = MARKETPLACE_URL + '/wp-json/wccom-extensions/2.0/featured';
+async function fetchDiscoverPageData(): Promise< ProductGroup[] > {
+	let url = '/wc/v1/marketplace/featured';
 
-	return fetch( fetchUrl )
-		.then( ( response ) => {
-			if ( ! response.ok ) {
-				throw new Error( response.statusText );
-			}
-			return response.json();
-		} )
-		.then( ( json ) => {
-			return json;
-		} )
-		.catch( () => {
-			return [];
-		} );
-};
+	if ( LOCALE.userLocale ) {
+		url = `${ url }?locale=${ LOCALE.userLocale }`;
+	}
+
+	try {
+		return await apiFetch( { path: url.toString() } );
+	} catch ( error ) {
+		return [];
+	}
+}
 
 function fetchCategories(): Promise< CategoryAPIItem[] > {
-	return fetch( MARKETPLACE_URL + '/wp-json/wccom-extensions/1.0/categories' )
+	let url = MARKETPLACE_URL + '/wp-json/wccom-extensions/1.0/categories';
+
+	if ( LOCALE.userLocale ) {
+		url = `${ url }?locale=${ LOCALE.userLocale }`;
+	}
+
+	return fetch( url.toString() )
```

Comment:
For consistency, wouldn't it make sense to apiFetch here? I mean, I guess it's not strictly necessary but we're using it above so maybe use it here too?

Reply:
Yeah, it would make sense and would make it more symmetrical. But couldn't make it work since `apiFetch` by default adds the `X-Wp-Nonce` header and totally breaks the WCCOM endpoint. There is no way to disable that behavior of `apiFetch` unfortunately. It always adds that nonce.

Some discussion: https://github.com/WordPress/gutenberg/discussions/46780